presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -878,6 +878,16 @@ protected RowExpression visitNullIfExpression(NullIfExpression node, Context con
                 // if (equal(cast(first as <common type>), cast(second as <common type>)), cast(null as firstType), first)
                 return specialForm(IF, returnType, equal, constantNull(originalFirst.getType()), originalFirst);
             }
+            else if (isNative) { // If both the types are same, but still native execution
```

Comment:
This branch seems to have some overlap with the previous branch. Consider refactoring.

Should we extend the tests for the original change to add coverage for NULLIF with same type arguments?

Reply:
Updated.